packages/expo-cli/src/commands/upload/submission-service/android/AndroidSubmitter.ts

```diff
@@ -0,0 +1,279 @@
+    const scheduleSpinner = ora('Scheduling submission').start();
+    let submissionId: string;
+    try {
+      submissionId = await SubmissionService.startSubmissionAsync(
+        Platform.ANDROID,
+        this.submissionConfig
+      );
+      scheduleSpinner.succeed();
+    } catch (err) {
+      scheduleSpinner.fail('Failed to schedule submission');
+      throw err;
+    }
+
+    let submissionCompleted = false;
+    let submissionStatus: SubmissionStatus | null = null;
+    let submission: Submission | null = null;
+    const submissionSpinner = ora('Submitting your app to Google Play Store').start();
+    try {
+      while (!submissionCompleted) {
+        await sleep(DEFAULT_CHECK_INTERVAL_MS);
+        submission = await SubmissionService.getSubmissionAsync(submissionId);
+        submissionSpinner.text = AndroidOnlineSubmitter.getStatusText(submission.status);
+        submissionStatus = submission.status;
+        if (submissionStatus === SubmissionStatus.ERRORED) {
+          submissionCompleted = true;
+          submissionSpinner.fail();
+        } else if (submissionStatus === SubmissionStatus.FINISHED) {
+          submissionCompleted = true;
+          submissionSpinner.succeed();
+        }
+      }
+    } catch (err) {
+      submissionSpinner.fail(AndroidOnlineSubmitter.getStatusText(SubmissionStatus.ERRORED));
+      throw err;
+    }
+
+    await displayLogs(submission, submissionStatus, this.verbose);
+  }
+
+  private static getStatusText(status: SubmissionStatus): string {
+    if (status === SubmissionStatus.IN_QUEUE) {
+      return 'Submitting your app to Google Play Store: waiting for an available submitter';
+    } else if (status === SubmissionStatus.IN_PROGRESS) {
+      return 'Submitting your app to Google Play Store: submission in progress';
+    } else if (status === SubmissionStatus.FINISHED) {
+      return 'Successfully submitted your app to Google Play Store!';
+    } else if (status === SubmissionStatus.ERRORED) {
+      return 'Something went wrong when submitting your app to Google Play Store.';
+    } else {
+      throw new Error('This should never happen');
+    }
+  }
+}
+
+interface Summary {
+  androidPackage: string;
+  archivePath?: string;
+  archiveUrl?: string;
+  archiveType: ArchiveType;
+  serviceAccountPath: string;
+  track: ReleaseTrack;
+  releaseStatus?: ReleaseStatus;
+  mode: SubmissionMode;
+}
+
+const SummaryHumanReadableKeys: Record<keyof Summary, string> = {
+  androidPackage: 'Android package',
+  archivePath: 'Archive path',
+  archiveUrl: 'Archive URL',
+  archiveType: 'Archive type',
+  serviceAccountPath: 'Google Service Account',
+  track: 'Release track',
+  releaseStatus: 'Release status',
+  mode: 'Submission mode',
+};
+
+const SummaryHumanReadableValues: Partial<Record<keyof Summary, Function>> = {
+  mode: (mode: SubmissionMode): string => {
+    if (mode === SubmissionMode.online) {
+      return 'Using Submission Service';
```

Comment:
```suggestion
      return 'Using Expo Submission Service';
```

usually not a fan of throwing the word Expo around everywhere but I think in this case it's worth clarifying that we are doing the submitting